### PR TITLE
Update AzureDisk SC volumeBindingMode to WaitForFirstConsumer

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:  Without this, PVC will stay Pending with message ` Warning  FailedScheduling        9m25s  default-scheduler                     0/3 nodes are available: 1 node(s) had taint {node-role.kubernetes.io/master: }, that the pod didn't tolerate, 2 node(s) had volume node affinity conflict.` because the disk won't be created in the same AZ that the pod is running in.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Update AzureDisk StorageClass volumeBindingMode to WaitForFirstConsumer
```

### CODE REVIEW

**Review by chewong — APPROVED** [review]
/lgtm

File: templates/addons/storageclass-azure-disk.yaml [hunk 1]
```diff
@@ -37,3 +39,4 @@ parameters:
   kind: Managed
   storageaccounttype: Standard_LRS
   cachingmode: ReadOnly
+volumeBindingMode: WaitForFirstConsumer
```
Comment (cpanato) [inline templates/addons/storageclass-azure-disk.yaml:42]: nit: new line not a blocker

**Review by cpanato — APPROVED** [review]
/lgtm

File: templates/addons/storageclass-azure-disk.yaml (hunk 1 above)
Comment (CecileRobertMichon) [reply templates/addons/storageclass-azure-disk.yaml]: added

**Comment by nader-ziada** [comment]
I ran into this issue as well and found this link https://kubernetes.io/docs/concepts/storage/storage-classes/#volume-binding-mode

/lgtm
/approve

**Pull request merged**